addons/purchase/models/purchase.py

```diff
@@ -706,25 +708,45 @@ def _confirm_reception_mail(self):
                 order.message_post(body="The order receipt has been acknowledged by %s." % order.partner_id.name)
 
     def _update_date_planned_for_lines(self, updated_dates):
-        if updated_dates:
-            note = self._compose_note(updated_dates)
-            self.activity_schedule(
+        # creat or update the activity
+        self._create_update_date_activity(updated_dates)
+        # update the date on PO line
+        for line, date in updated_dates:
+            line._update_date_planned(date)
```

Comment:
not very important but it'd seems to me updating the date planned should be done before the _create_update_date_activity

Reply:
It's actually important. Because if we update the date first, then we will lose the info about the old date when update the note in the activity.